### ISSUE TITLE
[web] TextField bug with IME

### ISSUE THREAD

**Issue by pedia** [flutter/flutter#119411]
## Steps to Reproduce

1. Execute `Run` on the code sample in [Dartpad](https://dartpad.dev/?id=5055057f46a3cf0a229532ffab0da3ee)
2. Focus in the only one text field
3. Turn on IME(pinyin method) and input `zh`

**Expected results:** 
<img width="453" alt="image" src="https://user-images.githubusercontent.com/170561/215259918-5fbbab7a-cf8c-42ce-b796-8efaa39e6b52.png">

**Actual results:** 
<img width="448" alt="image" src="https://user-images.githubusercontent.com/170561/215258958-dc8abd19-9b5f-41a6-9bc4-c78ade55fd4d.png">

The first letter of `z` is wrong. After some tests, I found `controller.text.isNotEmpty ?` is influence.

This only happened on `Web` platform with all channels(stable/master). It's OK on `macOS`.

**Comment by exaby73** [comment]
Hello @pedia. I get the following on MacOS:

![image](https://user-images.githubusercontent.com/50555895/215423852-04c32e40-d420-40d4-810c-2df9506f15b9.png)

And the following on Web:

![image](https://user-images.githubusercontent.com/50555895/215424005-1388bf7f-0f7d-49e3-a551-e37e0b1f7e34.png)

This is both on stable and master

**Comment by pedia** [comment]
@exaby73 The image of your paste are same as mine. 

It's wrong on Web. The first letter `z` is no effect to IME. 
`zh` should be same as
<img width="453" alt="image" src="https://user-images.githubusercontent.com/50555895/215423852-04c32e40-d420-40d4-810c-2df9506f15b9.png">

It is seems that the `TextField` ate `z`.

**Comment by exaby73** [comment]
Ahh thanks for the insight. 

## Triage report

As per https://github.com/flutter/flutter/issues/119411#issuecomment-1408171345, I can reproduce this issue on stable and master versions (3.7) with the steps outlined above. MacOS seems to work as expected but web doesn't.

<details><summary>Code Sample</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(
    const MaterialApp(
      debugShowCheckedModeBanner: false,
      home: DemoApp(),
    ),
  );
}

class DemoApp extends StatefulWidget {
  const DemoApp({super.key});

  @override
  State<DemoApp> createState() => _DemoAppState();
}

class _DemoAppState extends State<DemoApp> {
  final controller = TextEditingController();

  @override
  void initState() {
    super.initState();
    controller.addListener(_foo);
  }

  @override
  void dispose() {
    controller.removeListener(_foo);
    controller.dispose();
    super.dispose();
  }

  void _foo() {
    setState(() {});
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Padding(
        padding: const EdgeInsets.all(32),
        child: Column(
          children: const [TextField()],
        ),
      ),
    );
  }
}
```

</details>

</details>

<details><summary><code>flutter doctor -v</code> (Master)</summary>

```
[!] Flutter (Channel master, 3.7.0-31.0.pre.24, on macOS 13.1 22C65 darwin-arm64, locale en-US)
    • Flutter version 3.7.0-31.0.pre.24 on channel master at /Users/nabeelparkar/fvm/versions/master
    ! Warning: `dart` on your path resolves to /opt/homebrew/Cellar/dart/2.18.6/libexec/bin/dart, which is not inside your current Flutter SDK checkout at /Users/nabeelparkar/fvm/versions/master. Consider adding /Users/nabeelparkar/fvm/versions/master/bin to the front of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 7ba440655a (2 hours ago), 2023-01-30 20:37:21 -0800
    • Engine revision 9448f2966c
    • Dart version 3.0.0 (build 3.0.0-179.0.dev)
    • DevTools version 2.20.1
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git" directly to perform update checks and upgrades.

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
    • Android SDK at /Users/nabeelparkar/Library/Android/sdk/
    • Platform android-33, build-tools 33.0.0
    • ANDROID_SDK_ROOT = /Users/nabeelparkar/Library/Android/sdk/
    • Java binary at: /Users/nabeelparkar/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/213.7172.25.2113.9123335/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14C18
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • CHROME_EXECUTABLE = /Applications/Brave Browser.app/Contents/MacOS/Brave Browser

[✓] Android Studio (version 2021.3)
    • Android Studio at /Users/nabeelparkar/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/213.7172.25.2113.9123335/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)

[!] Android Studio (version 2022.1)
    • Android Studio at /Users/nabeelparkar/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/221.6008.13.2211.9477386/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    ✗ Unable to find bundled Java version.
    • Try updating or re-installing Android Studio.

[✓] VS Code (version 1.74.3)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.58.0

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-arm64   • macOS 13.1 22C65 darwin-arm64
    • Chrome (web)    • chrome • web-javascript • Brave Browser 109.1.47.186

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 2 categories.
```

</details>

**Comment by pedia** [comment]
Confirm resolved
